### ISSUE TITLE
DQNAgent convergance ?

### ISSUE THREAD

**Issue by ganarajpr** [uaca/deepy#14]
Hi,

I am doing my own experiments with the DQNAgent - the only difference is that I want it to go to the target location and stop. 

In contrast, in the puckworld, the agent is continuously looking for new "fruits" to gobble up. In my version, I want it to find one "fruit" in the search space with the most efficient path ( set of actions! ). The agent keeps running without converging on the point that provides the highest reward. Is there anything that I need to change to get this working that way ?

**Comment by zomux** [comment]
Hi,

Well, There is a hyper-parameter (EPSILON in deepy/experiments/deep_qlearning/agent.py) to control the probability to perform a random action.

You can set EPSILON to zero, and see how it goes.

If it does not work, the network may has some underfitting / overfitting problem, may be some hyper-parameters like weight_l2 should be increased.

Basically, when I did the experiment, I found my model can collect more rewards than that in http://cs.stanford.edu/people/karpathy/reinforcejs/puckworld.html, but it's behavior is unstable as you said.

Please tell me if you find anything about it.

**Comment by zomux** [comment]
I updated the trained deep Q agent model, I don't know whether it solves the problem, you can try again

```
git pull
```
